.github/workflows/ci.yml

```
@@ -102,15 +102,11 @@ jobs:
       with:
         tool: cargo-llvm-cov@0.4.5
     - name: Setting up cache
-      uses: pl-strflt/rust-sccache-action@v1
-      env:
-        SCCACHE_CACHE_SIZE: 2G
-        SCCACHE_DIR: ${{ github.workspace }}/.cache/sccache
-        CACHE_SKIP_SAVE: ${{ matrix.push == '' || matrix.push == 'false' }}
+      uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f
```

FYI, I removed the deprecated GitHub Actions usage (https://github.com/pl-strflt/rust-sccache-action/pull/16) and added autoclean feature (https://github.com/pl-strflt/rust-sccache-action/pull/18) to `rust-sccache-action`.

So if we wanted to stick with incremental builds we could go with:

```suggestion
      uses: pl-strflt/rust-sccache-action@v1
      env:
        SCCACHE_CACHE_SIZE: 2G
        SCCACHE_DIR: ${{ github.workspace }}/.cache/sccache
      with:
        version: v0.2.15
        # change this to invalidate sccache for this job
        shared-key: ${{ matrix.key }}
        skip-save: ${{ matrix.push == '' || matrix.push == 'false' }}
        autoclean: true
```

Otherwise, this looks good 👍

---

Ah, wonderful. I'll try that first.

---

Ah, nevermind. That's what we were already doing, but it doesn't seem to _actually_ keep the cache clean. It just grows and grows.